src/main/java/com/develhope/spring/order/repository/OrderRepository.java

```diff
@@ -4,6 +4,11 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.Optional;
+
 @Repository
 public interface OrderRepository extends JpaRepository <Order, Long> {
+    boolean existsByUserId(long id);
```

Comment:
Questo metodo non serve visto che abbiamo già un existsById di default, non c'è bisogno che vai a controllare il caso in cui non esista il suo userId perchè altrimenti sarebbe un problema della richiesta post (dove hai invece già trattato alcuni casi di eccezione)